src/calendar/calendar.tsx

```diff
@@ -0,0 +1,348 @@
+    // =============================================================================
+
+    const generateWeeksOfTheMonth = useMemo(
+        (): Dayjs[][] => generateCalendar(calendarDate),
+        [calendarDate]
+    );
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const YearHeaderComponent = () => {
+        if (showView === "Year") {
+            const beginDecaded =
+                Math.floor(+calendarDate.format("YYYY") / 10) * 10;
+
+            const endDecaded = dayjs(`${beginDecaded}-01-01`)
+                .add(9, "year")
+                .format("YYYY");
+
+            return (
+                <Text.BodySmall>
+                    {beginDecaded} to {endDecaded}
+                </Text.BodySmall>
+            );
+        }
+
+        return (
+            <Text.BodySmall>
+                {dayjs(calendarDate).format("YYYY")}
+            </Text.BodySmall>
+        );
+    };
+
+    return (
+        <Container>
+            <HeaderDropdown view={showView}>
+                <DropdownMonth onClick={() => handleView("Month")}>
+                    <Text.BodySmall>
+                        {dayjs(calendarDate).format("MMM")}
+                    </Text.BodySmall>
+                    <ChevronDownIcon />
+                </DropdownMonth>
+                <DropdownYear onClick={() => handleView("Year")}>
+                    {YearHeaderComponent()}
+                    <ChevronDownIcon />
+                </DropdownYear>
+            </HeaderDropdown>
+            <Views show={showView}>
+                <DayView>
+                    <WeekDaysContainer>
+                        {generateWeeksOfTheMonth[0].map((day, index) => (
+                            <WeekCell key={`week-day-${index}`}>
+                                {dayjs(day).format("ddd")}
+                            </WeekCell>
+                        ))}
+                    </WeekDaysContainer>
+                    {generateWeeksOfTheMonth.map((week, weekIndex) => (
+                        <CalendarDaySection key={`week-${weekIndex}`}>
+                            {week.map((day, dayIndex) => {
+                                const {
+                                    dateStartWithYear,
+                                    isDisabled,
+                                    variant,
+                                } = generateDayStatus(day);
+
+                                const classNames =
+                                    generateDayClass(dateStartWithYear);
+
+                                return (
+                                    <GrowDayCell
+                                        key={`day-${dayIndex}`}
+                                        className={classNames}
+                                    >
+                                        <LeftCell data-cell="left">
```

Comment:
are these `data-cell` and `data-circle` attributes used for anything?

Reply:
`data-cell` and `data-circle` use in css styling

Reply:
Updated: 
1. Namespace and Naming for calendar-helper
2. Move `generateFirstDayOfEachWeek` and `generateWeek` outside of the main fn in calendar-helper file.
3. Remove `clone()` for dayjs
4. Create new component for `CalendarDay`
5. All `px` changed to `rem`
6. Use `Dayjs` in month and year component instead of `Date`
7. Remove `null` in initial mount in month calendar
8. Re-name to `nextDecade ` from `nextDecaded`
9. Move `generateMonth` and `generateDecadeOfYear` into calendar-helper
10. Added unit test for `generateMonth` and `generateDecadeOfYear` in calendar-helper

Pending:
1. Simplify for dayjs(calendarDate).month(i)
2. Simplify for dayjs(calendarDate).year(i)